src/irmin/commit.ml

```diff
@@ -179,10 +188,15 @@ module History (S : S.COMMIT_STORE) = struct
     in
     let min = List.map (fun k -> `Commit k) min in
     let max = List.map (fun k -> `Commit k) max in
-    Graph.closure ~pred ~min ~max () >|= fun g ->
-    List.fold_left
-      (fun acc -> function `Commit k -> k :: acc | _ -> acc)
-      [] (Graph.vertex g)
+    Graph.closure ~pred ~min ~max () >>= function
+    | Error e -> closure_error e
+    | Ok g ->
+        let keys =
+          List.fold_left
+            (fun acc -> function `Commit k -> k :: acc | _ -> acc)
+            [] (Graph.vertex g)
+        in
+        Lwt.return_ok keys
```

Comment:
I was trying to make it uniform with the similar functions `closure_error` in `irmin/store.ml` but there is probably no need for that.

Reply:
Indeed, I think it's better to use `map` here. It doesn't matter much, but as far as I understand `map f` is faster than `bind (f >> return)` since it's not necessary to check if the result of `f` is resolved.